### PR TITLE
Java test of XSL stylesheet

### PR DESCRIPTION
**Pull request by g4s8**
#255 - xsl validator app with Saxon library

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#284](https://github.com/zerocracy/datum/issues/284) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#284](https://github.com/zerocracy/datum/issues/284) is assigned to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), here is [why](https://www.0crat.com/footprint/C3NDPUA8L/480923742). The budget is 15 minutes, see [§4](http://datum.zerocracy.com/pages/policy.html#4). Please, read [§27](http://datum.zerocracy.com/pages/policy.html#27) and when you decide to accept the changes, inform @yegor256[/z](https://www.0crat.com/u/yegor256) (the architect) right in this ticket. If you decide that this PR should not be accepted ever, also inform the architect.

**Comment by 0crat** [comment]
@carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) this job was assigned to you 5 days ago. It will be taken away from you soon, unless you close it, see [§8](http://datum.zerocracy.com/pages/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

File: validator/src/test/java/com/zerocracy/validator/XmlTransformedTest.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+ * Test case for {@link XmlTransformed}.
+ *
+ * @author Kirill (g4s8.public@gmail.com)
+ * @version $Id$
+ * @since 0.1
+ * @checkstyle JavadocMethodCheck (500 lines)
+ */
+public final class XmlTransformedTest {
+    @Test
+    public void transformXmlWithXsl() throws Exception {
```
Comment (carlosmiranda) [inline validator/src/test/java/com/zerocracy/validator/XmlTransformedTest.java]: @g4s8 no javadoc for this test?

Comment (g4s8) [reply validator/src/test/java/com/zerocracy/validator/XmlTransformedTest.java]: @carlosmiranda as I know almost all test methods in `farm` project doesn't have javadocs, so why to add it here?

**Comment by 0crat** [comment]
@carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) this job was assigned to you 8 days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by carlosmiranda** [comment]
@yegor256 looks good to merge

**Comment by carlosmiranda** [comment]
@0crat waiting

**Comment by 0crat** [comment]
> @0crat waiting ([here](https://github.com/zerocracy/datum/issues/284#issuecomment-368356918))

@carlosmiranda It's a code review job [#284](https://github.com/zerocracy/datum/issues/284), can't put it on hold

**Comment by yegor256** [comment]
@g4s8 I'm surprised that you didn't catch the bug we have in `equity.xsl`. 0crat.com crashes now at that file, but your tests don't see the bug. Why?

**Comment by 0crat** [comment]
The user @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda) resigned from [#284](https://github.com/zerocracy/datum/issues/284), please stop working. Reason for job resignation: It is older than 10 days, see [§8](http://www.zerocracy.com/policy.html#8)

**Comment by 0crat** [comment]
Resigned on delay, see [§8](http://www.zerocracy.com/policy.html#8): -15 points just awarded to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), total is [+1650](https://www.0crat.com/u/carlosmiranda)

**Comment by 0crat** [comment]
Mistake of @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda)[/z](https://www.0crat.com/u/carlosmiranda) (your student): Resigned on delay, see [§8](http://www.zerocracy.com/policy.html#8): -15 points just awarded to @yegor256[/z](https://www.0crat.com/u/yegor256), total is [+10959](https://www.0crat.com/u/yegor256)

**Comment by g4s8** [comment]
@yegor256 see new commit, I reproduced a bug from https://github.com/zerocracy/farm/issues/468 issue:
```
java.lang.IllegalStateException: failed to configure transformer by net.sf.saxon.TransformerFactoryImpl

	at com.jcabi.xml.XSLDocument.transformInto(XSLDocument.java:413)
	at com.jcabi.xml.XSLDocument.transform(XSLDocument.java:368)
	at com.zerocracy.validator.XmlTransformed.value(XmlTransformed.java:52)
	at com.zerocracy.validator.XmlTransformedTest.transformXmlWithXsl(XmlTransformedTest.java:48)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)
Caused by: javax.xml.transform.TransformerConfigurationException: Failed to compile stylesheet. 1 error detected.
	at net.sf.saxon.PreparedStylesheet.prepare(PreparedStylesheet.java:176)
	at net.sf.saxon.TransformerFactoryImpl.newTemplates(TransformerFactoryImpl.java:139)
	at net.sf.saxon.TransformerFactoryImpl.newTransformer(TransformerFactoryImpl.java:91)
	at com.jcabi.xml.XSLDocument.transformInto(XSLDocument.java:402)
	... 25 more
```

**Comment by yegor256** [comment]
@g4s8 wait, but why did you make duplicates of the files? The bug is in `xsl/` directory. Your tests should catch it and complain. Then, you will have to fix the bug in our production files. Don't copy them. The point of this testing framework is to validate the existing files.

**Comment by g4s8** [comment]
@yegor256 because this test app is not integrated yet, I created a `todo` to run it against files in xsl directory

**Comment by yegor256** [comment]
@rultor merge

**Comment by 0crat** [comment]
The job [#284](https://github.com/zerocracy/datum/issues/284) is now out of scope

**Pull request merged**